core/node/db_pruner/src/lib.rs

```diff
@@ -290,7 +317,9 @@ impl DbPruner {
                     self.health_updater.update(health);
                     true
                 }
-                Ok(pruning_done) => !pruning_done,
+                Ok(PruningIterationOutcome::Interrupted) => break,
```

Comment:
wouldn't the pruner have stopped anyway even without this additional logic of PruningIterationOutcome?

Reply:
It would, but IMO returning `PruningIterationOutcome` (i.e., a specialized type vs `bool`) makes it cleaner.

Reply:
well, but you didn't just add a type for "pruning_done", you also added third outcome and assert for specific outcome in tests.

Reply:
IMHO it's either "be more explicit" or "have more branching". Either option is fine, for me.